File: client/src/lib/types/job.ts
/**
 * Job-related type definitions
 */

// Backend Job schema (matches Go struct)
export interface Job {
  id: string;
  title: string;
  companyID: string;
  location: string;
  workType: string;
  workArrangement: string;
  currency: string;
  minSalary: number;
  maxSalary: number;
  jobDescription: string;
  jobSummary: string;
  requiredSkills: string;
  experienceLevel: string;
  education: string;
  niceToHave?: string;
  questions?: string;
  postOpenDate: string;
  applicationDeadline: string;
  numberOfPositions: number;
  visibility: string;
  emailNotifications: boolean;
  autoReject: boolean;
}

// Frontend-enhanced job (for display purposes)
export interface JobDisplay extends Job {
  company?: string;
  companyLogo?: string;
  status?: string;
  applicants?: number;
  posted?: string;
  expires?: string;
}

// Query filters (matches backend query params)
export interface JobFilters {
  id?: string;
  title?: string;
  companyID?: string;
  location?: string;
  minSalary?: number;
  maxSalary?: number;
  workType?: string;
  workArrangement?: string;
  postOpenDate?: string; // "1d" | "6w"
  latest?: boolean;
  sort?: "dateAsc" | "dateDesc" | "title";
}

export interface DeleteJobRequest {
  reason: string;
}


// Form-related types
export interface JobFormData {
  // Basic Info
  jobTitle?: string;
  companyID?: string;
  location?: string;
  workType?: string;
  workArrangement?: string;
  currency?: string;
  minSalary?: number | string;
  maxSalary?: number | string;
  
  // Description
  jobDescription?: string;
  jobSummary?: string;
  
  // Requirements
  requiredSkills?: string[];
  yearsOfExperience?: string;
  educationLevel?: string;
  
  // Post Settings
  postingOpenDate?: string;
  postingCloseDate?: string;
  screeningQuestions?: string;
  emailNotifications?: boolean;
  
  // Application Requirements
  applicationRequirements?: {
    resume?: boolean;
    coverLetter?: boolean;
    portfolio?: boolean;
    linkedin?: boolean;
  };
}

export interface ValidationState {
  errors: Record<string, string>;
  showErrors: boolean;
}

File: server/internal/dto/job.go
package dto

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Job struct {
	ID                  *primitive.ObjectID `bson:"_id,omitempty" json:"id,omitempty"`
	Title               *string             `bson:"title,omitempty" json:"title,omitempty"`
	Location            *string             `bson:"location,omitempty" json:"location,omitempty"`
	WorkType            *string             `bson:"workType,omitempty" json:"workType,omitempty"`
	WorkArrangement     *string             `bson:"workArrangement,omitempty" json:"workArrangement,omitempty"`
	Currency            *string             `bson:"currency,omitempty" json:"currency,omitempty"`
	MinSalary           *float64            `bson:"minSalary,omitempty" json:"minSalary,omitempty"`
	MaxSalary           *float64            `bson:"maxSalary,omitempty" json:"maxSalary,omitempty"`
	JobDescription      *string             `bson:"jobDescription,omitempty" json:"jobDescription,omitempty"`
	JobSummary          *string             `bson:"jobSummary,omitempty" json:"jobSummary,omitempty"`
	RequiredSkills      *string             `bson:"requiredSkills,omitempty" json:"requiredSkills,omitempty"`
	ExperienceLevel     *string             `bson:"experienceLevel,omitempty" json:"experienceLevel,omitempty"`
	Education           *string             `bson:"education,omitempty" json:"education,omitempty"`
	NiceToHave          *string             `bson:"niceToHave,omitempty" json:"niceToHave,omitempty"`
	Questions           *string             `bson:"questions,omitempty" json:"questions,omitempty"`
	PostOpenDate        *time.Time          `bson:"postOpenDate,omitempty" json:"postOpenDate,omitempty"`
	ApplicationDeadline *time.Time          `bson:"applicationDeadline,omitempty" json:"applicationDeadline,omitempty"`
	NumberOfPositions   *int                `bson:"numberOfPositions,omitempty" json:"numberOfPositions,omitempty"`
	Visibility          *string             `bson:"visibility,omitempty" json:"visibility,omitempty"`
	EmailNotifications  *bool               `bson:"emailNotifications,omitempty" json:"emailNotifications,omitempty"`
	AutoReject          *bool               `bson:"autoReject,omitempty" json:"autoReject,omitempty"`
}
